src/ftsh.c

```diff
@@ -0,0 +1,119 @@
+/***************************************************************************//**
+
+	@file         ftsh.c
+
+	@author       Max Bronckers
+
+	@date         Thursday, 1 May 2018
+
+	@brief        FTSH (Full-Text SHell)
+
+	@compile      make ftsh
+
+*******************************************************************************/
+
+#include "ftsh.h"
+#include "ftsh_functions.h"
+
+/**
+	@brief Parse input into args.
+	@param input The input.
+	@return Null-terminated array of args.
+ */
+char** ftsh_get_input(char *input) {
+	char **args = malloc(8 * sizeof(char *));
+	char *separator = " ";
+	char *parsed;
+	int index = 0;
+
+	/* Error handling */
+	if (args == NULL) {
+		perror("ftsh malloc failed");
+		exit(1);
+	}
+
+	/* Exit on Ctrl-D */
+	if (input == NULL) {  
+      	printf("\n");
+      	exit(0);
+    }
+
+	parsed = strtok(input, separator);
+
+	while (parsed != NULL) {
+		args[index++] = parsed;
+		parsed = strtok(NULL, separator); 
+	}
+
+	args[index] = NULL;
+	return args;
+}
+
+/**
+	@brief Launch a program and wait for it to terminate.
+	@param args Null terminated list of arguments (including program).
+	@return Always returns 1, to continue execution.
+ */
+int ftsh_launch(char **args)
+{
+	pid_t pid;
+	int status;
+
+	pid = fork();
+	if (pid == 0) {
+		// Child process
+		if (execvp(args[0], args) == -1) {
```

Comment:
This is a lovely implementation of the CS154 shell project, but that's not what we're asking you to do. From the project specification:

> The full text search tool must allow a user to open a text file, to specify a word or words, and to interactively navigate the text file, skipping from matched word to matched word.

Nothing in there mandates a full UNIX shell, and you should not be implementing one. All that's necessary is a simple `while` loop that handles *only* the commands necessary for this project.

Reply:
Oh, Lydia had provided another team with a link to a tutorial to build a shell, so I assumed we were supposed to do the same. Will fix and simplify.